### PR TITLE
Add a META file to support ocamlfind linking.

### PR DESCRIPTION
**Pull request by ejgallego**
This is a preliminary version, comments are welcome !

It is not "mature" yet, on the other hand, merging it cannot do any harm.

### CODE REVIEW

File: META [hunk 1]
```diff
@@ -0,0 +1,36 @@
+description = "Coq Proof Assistant Plugin API"
+version = "dev"
+
+requires = "camlp5"
```
Comment (silene) [inline META]: This seems fishy, since camlp4 is also supported. I guess META should be (partly) generated by configure.

Comment (ejgallego) [inline META]: Indeed this META file is barely a sketch, but for that particular purpose I wouldn't support camlp4 anymore.

By the way, I would avoid automatic generation of META files. `exists_if` and other conditional constructs exist there to handle variable cases (for instance you can do:
```
package "ide" (
    description = "Coq Ide Libraies"
    version = "[distributed with Coq]"
    exists_if = "ide/idelib.cmxa"
...
}

**Comment by ejgallego** [comment]
IMO this is ready to merge now.

**Comment by ejgallego** [comment]
It is missing however the install target, maybe @letouzey knows what to do here?

**Comment by ejgallego** [comment]
The install target is working now, it'd be great if this could make its way to OPAM. (As Coq ships no opam file, I dunno if I can pin my local directory to try).

**Review by mattam82 — APPROVED** [review]
Great, it seems more thought out than any ocamlfind setup I've seen before, so I'm happy to start from this even if it's not entirely polished.

**Comment by ejgallego** [comment]
Cool, thanks for the review. Note the current setup is enough to build ML plugins and toplevel with just ocamlbuild, which is very convenient. There's an example at https://github.com/ejgallego/coq-serapi/blob/coq-ocamlfind/_tags

Indeed the above illustrates a situation where `coq_makefile` falls short: integrating Coq with many OPAM packages.

**Comment by ejgallego** [comment]
I should also note that this allows building coqide and coqcheck with ocamlbuild, which removes many complexities from the build system.

**Comment by mattam82** [comment]
Yep, it's especially useful when developing plugins as well!

**Comment by Zimmi48** [comment]
Maybe the documentation should be updated as well to explain what this allows to do.

**Comment by ejgallego** [comment]
@Zimmi48 that is a good point, where would it go? There is something like an official example plugin/official plugin writing guide? Basically you can add packages coq.ltac etc...

**Comment by Zimmi48** [comment]
@ejgallego: there is no such official tutorial as far as I know but as I understand it, official documentation for developers is supposed to go in `dev/doc`. There is a file called `extensions.txt` in there but it is still in French unfortunately. Gregory Malecha has a plugin template here: https://github.com/gmalecha/coq-plugin-template

**Comment by ejgallego** [comment]
An important shortcoming of `ocamlbuild` is that it doesn't support `.v` files. I propose we use as documentation some of the example projects using ocamlbuild such as SerAPI or the split CoqIDE https://github.com/ejgallego/coqide-exp/blob/master/_tags (need to port it thou)

**Comment by maximedenes** [comment]
I'll remove the path referencing your home, Emilio (it is in a comment, not in actual code) and merge it.

**Comment by maximedenes** [comment]
Merged, thanks a lot!

**Pull request closed without merging**